### ISSUE TITLE
Using RubyMineReporter and HtmlReporter does not retain HTML results

### ISSUE THREAD

**Issue by icantrap** [kern/minitest-reporters#226]
If you run `rake test` with both RubyMine and HTML reporters, you get a HTML report, as expected. If you subsequently run a test in RubyMine, the HTML report is wiped out without writing a new HTML report. Shouldn't it either leave the old report intact or write a new HTML report?

I created a sample project to illustrate this - https://github.com/icantrap/minitest-reporters-sample.

Am I setting up the reporters incorrectly?

**Comment by os97673** [comment]
I see nothing special in the way you configure reporters, so, I'd suggest to contact RubyMine team to check if they consider this an issue: you can either file ticket at https://youtrack.jetbrains.com/issues/RUBY or ask on their forum (https://intellij-support.jetbrains.com/hc/en-us/community/topics/200366799-RubyMine)

I'm closing the issue for now but feel free to ask more questions our point them to me.

**Comment by icantrap** [comment]
Actually, I went through your code and I figured out the root cause.

The HTML report is erased when the `HtmlReporter` is initialized. Then, when the `use!` call is made, if it's a RubyMine environment, `Minitest::Reporters#choose_reporters` will eventually discard the configured reporter array and replace it with an array with a single new `RubyMineReporter`. That's how the HTML report gets erased, but the new report is not written.

Moving the erasure of the HTML report to the `start` hook allows the report to survive the replacement of the configured reporter array.

I have a fork that lets me get pass this problem. I'll submit a PR, if you're interested.

Thanks for your assistance.
